src/main/scala/com.gu.gibbons/config/Settings.scala

```diff
@@ -32,26 +35,25 @@ object Settings {
     "ea39a2bb-630d-4565-97ef-a47eff4ec300"
   )
 
-  def fromEnvironment: Either[String, Settings] = {
-    val env = System.getenv.asScala.toMap
-    for{
-      region <- getEnv(env, "AWS_REGION")
-      salt <- getEnv(env, "SALT")
-      origin <- getEnv(env, "EMAIL_ORIGIN")
-      bonoboUrl <- getEnv(env, "BONOBO_URL")
-      usersTableName <- getEnv(env, "BONOBO_USERS_TABLE")
-    } yield {
-      Settings(
-        region = Regions.fromName(region),
-        usersTableName = usersTableName, 
-        salt = salt,
-        bonoboUrl = bonoboUrl,
-        fromAddress = Email(origin),
-      )
+  def fromEnvironment: ValidatedNel[String, Settings] =
+    parseEnv(System.getenv.asScala.toMap)
+    
+  def parseEnv(env: Map[String, String]) =
+    ( getEnv(env, "AWS_REGION").andThen(makeRegion)
+    , getEnv(env, "BONOBO_USERS_TABLE")
+    , getEnv(env, "SALT")
+    , getEnv(env, "BONOBO_URL")
+    , getEnv(env, "EMAIL_ORIGIN").map(Email(_))
+    ).mapN(Settings(_, _, _, _, _))
```

Comment:
I can't decide if this is brilliant or horrible

Reply:
I should be able to just write `mapN(Settings)` but for some reason it won't compile.

Reply:
... anyway it is a straight mapping of order _n_ (hence the name 😅 ) so I think it's fine.